### PR TITLE
Meta: CONTRIBUTING.md: Some changes to make the doc more beginner-friendly

### PR DESCRIPTION
**Pull request by tomayac**
## Editorial changes

- Differentiate "Issue" (uppercase, as in the GitHub concept) from "issue" (lowercase, the problem). Same for "Pull Request".
- Link the mailing list.
- Minor word changes.

## Layout and markup

- Use code font for file names and branch names (Are you considering switching from `master` to `main`?).
- Add new line after headings.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (ljharb) [inline CONTRIBUTING.md:49]: es-discuss is shut down; we should be pointing people at es.discourse.group

**Review by ljharb — COMMENTED** [review]
I’m not sure capitalizing “issue” makes much sense; this is GitHub, so I’d think a non-GitHub-issue “issue” occurrence should use a synonym instead of trying to differentiate it from GitHub issues

File: CONTRIBUTING.md
Comment (tomayac) [reply CONTRIBUTING.md]: Replaced with https://es.discourse.group/

**Review by ljharb — COMMENTED** [review]
LGTM

File: CONTRIBUTING.md
Comment (michaelficarra) [inline CONTRIBUTING.md:63]: As a reader, I would have no idea what "the discussion group" means.

Comment (tomayac) [inline CONTRIBUTING.md:63]:
```suggestion
To track what's going on with a particular proposal, you can look in issues and commits in the individual proposal repository, read presentation slides which are linked from the TC39 agenda, and read the notes which came from the subsequent meetings. You can also reach out via IRC, the [discussion group](https://es.discourse.group/), or direct communication with a proposal champion, if the other resources are unclear.
```

Comment (tomayac) [reply CONTRIBUTING.md]: It was linked above, but linking it again now.

File: CONTRIBUTING.md
Comment (michaelficarra) [inline CONTRIBUTING.md:49]: The discourse group and es-discuss are different things. Are we dropping references to es-discuss?

Comment (tomayac) [reply CONTRIBUTING.md:49]: I was [told](https://github.com/tc39/ecma262/pull/2431#discussion_r649227642) by @ljharb that the list is dead.

Comment (michaelficarra) [reply CONTRIBUTING.md:49]: That's fine, there's just been controversy within committee when we've previously swapped out references to es-discuss with references to the Discourse group. We might want a chair's approval on this PR before merging. @tc39/chairs

File: CONTRIBUTING.md
Comment (syg) [inline CONTRIBUTING.md:63]: The IRC bit might be worth updating with our migration to Matrix. Can be done in separate PR and doesn't need to block this PR.

**Review by syg — APPROVED** [review]
lgtm

File: CONTRIBUTING.md
Comment (ljharb) [reply CONTRIBUTING.md:49]: The controversy was because es-discuss was still usable. It is no longer writable, so it wouldn’t make sense to refer to it here anymore.

**Comment by tomayac** [comment]
Looks like we have 2 LGTMs then, but @ljharb's wasn't properly registered since it's a [comment](https://github.com/tc39/ecma262/pull/2431#pullrequestreview-681866679).

**Comment by ljharb** [comment]
@tomayac that was intentional, as I’m not an editor. It will be landed when it gets sufficient reviews.

**Review by michaelficarra — APPROVED** [review]

**Pull request merged**
